### ISSUE TITLE
Question: Create new table in python or ruby

### ISSUE THREAD

**Issue by blackfist** [facebook/osquery#1006]
Is C++ the only option for creating new tables in osquery? The documentation seems to suggest that it could be done, but I would really like to see an example. I'm trying to do it in ruby but I speak python better so either one is fine.

I *think* the way it works is to launch osqueryd with the `--extensions_socket` option and point it at some socket. 

Then I did a `thrift -r --gen rb osquery.thrift` to make some ruby code. That gave me bindings that I was able to use to connect to the socket and run queries and stuff.

I think I need to write a program that will do what I want and return that as a table and I need to register the table. That's the part where I could really use an example. Or somebody to tell me "No you're all wrong and you're mixing X up with Y."

If this turns out to be something that is not well documented I will volunteer to write the documentation once I have some working examples made.

**Comment by theopolis** [comment]
That's exactly how it should be done! The [extensions docs](http://osquery.readthedocs.org/development/osquery-sdk/) need some work and examples.

But... the "register" API wont do much but route calls to your external plugins (of which one could be a new table). The target language, python or ruby, would need to implement the [RegistryFactory](https://github.com/facebook/osquery/blob/master/include/osquery/registry.h#L453) class. In C++ that class takes requests using plugin names, finds the appropriate object, and calls the requested action. It's not so straight forward to write a similar router for another language. 

If we spent some time on writing python or ruby bindings it would give you the exact workflow that we describe in the C++ main function [here](http://osquery.readthedocs.org/development/osquery-sdk/#extensions).

**Comment by marpaia** [comment]
@blackfist, I took a run at a *crude* python extension, using just the thrift APIs: https://gist.github.com/marpaia/ddb09d4029a177850db3

It's obviously not high quality and shouldn't be used as any sort of a standard. I was just talking with @theopolis earlier today about how we want to design/package/distribute the `pyosquery` module, so stay tuned for that.

In the meanwhile, hopefully that gist helps nail down the minimum viable set of things that need to happen for an extension to work. For example, I ran osqueryd with this schedule:

```json
{
  "schedule": {
    "from_ext": {
      "query": "select * from from_extension",
      "interval": 2
    }
  }
}
```

And got the following log:

```json
{"name":"from_ext","hostIdentifier":"vagrant-ubuntu-trusty-64","calendarTime":"Mon Apr 20 03:54:06 2015","unixTime":"1429502046","columns":{"baz":"baz","foo":"bar"},"action":"added"}
```
